eea/facetednavigation/widgets/widget.py

```diff
@@ -235,26 +237,43 @@ def portal_vocabulary(self):
         """Look up selected vocabulary from portal_vocabulary or from ZTK
            zope-vocabulary factory.
         """
-        vtool = getToolByName(self.context, 'portal_vocabularies', None)
         voc_id = self.data.get('vocabulary', None)
         if not voc_id:
             return []
-        voc = getattr(vtool, voc_id, None)
-        if not voc:
-            voc = queryUtility(IVocabularyFactory, voc_id, None)
```

Comment:
collective.taxonomy is already using IVocabularyFactory mechanism. No need to hard-dependency collective.taxonomy